core/modules/field/modules/text/text.module

```diff
@@ -462,8 +462,8 @@ function text_summary($text, $format_id = NULL, $size = 600) {
   // If no complete paragraph then treat line breaks as paragraphs.
   $line_breaks = array('<br />' => 6, '<br>' => 4);
   // Newline only indicates a line break if line break converter
-  // filter is present.
-  if (isset($format->filters['filter_autop'])) {
+  // filter is enabled.
+  if (isset($format->filters['filter_autop']) && ($format->filters['filter_autop']->status != 0)) {
```

Comment:
The second condition could get simplified a bit.
The parenthesis aren't really necessary, and it would be sufficient to only check for a simple truthy value.

Like: `if (isset($format->filters['filter_autop']) && $format->filters['filter_autop']->status) {`

That it's a number has to do with Backdrop form API (converting checkbox values to 0/1).